cmake/cores/Cortex-A9.cmake

```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2020 ARM Limited. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+if(MBED_OS_TOOLCHAIN strequal "GCC_ARM")
```

Comment:
The profile cmake files said "MBED_TOOLCHAIN". Should we use "MBED_TOOLCHAIN" or "MBED_OS_TOOLCHAIN"? They should match, whichever we choose, right?

Reply:
probably we shall use `MBED_` as we use anywhere else - function or macro, always `MBED_` (don't need OS there) and its specific enough. We could fix the rest of the cmakes if that makes sense.

Reply:
OK, so we should do this for this PR:

```diff
- MBED_OS_TOOLCHAIN
+ MBED_TOOLCHAIN
```

to match the CPU cmake files.